[jit][redo] move casting ops from prim to aten

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22276 [jit] register __getitem__ builtin
* **#22275 [jit][redo] move casting ops from prim to aten**

Summary:

This is a redo PR of https://github.com/pytorch/pytorch/pull/21002, the previous one somehow get reverted because of a master breakage, submit it again. 

Currently we are messed up with prim and aten namespace when we register prim and builtin ops, lots of the ops that have schema should go into the aten namespace rather than prim, prim namespace should only be reserved for the ops that does not have a schema.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D16060597](https://our.internmc.facebook.com/intern/diff/D16060597)

wanchaol committed:
[jit][redo] move casting ops from prim to aten

Summary:

This is a redo PR of https://github.com/pytorch/pytorch/pull/21002

Currently we are messed up with prim and aten namespace when we register prim and builtin ops, lots of the ops that have schema should go into the aten namespace rather than prim, prim namespace should only be reserved for the ops that does not have a schema.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

zdevito commented on aten/src/ATen/core/interned_strings.h:
```diff
@@ -65,10 +65,10 @@ namespace c10 {
+  _(aten, Bool)                    \
+  _(aten, Int)                     \
+  _(aten, Float)                   \
+  _(aten, Str)                     \
```
Please don't  capitalize, it is suppose to match Python. Note that Bool/Int/Float have to be different because they would collide with C keywords

wanchaol committed:
Update on "[jit][redo] move casting ops from prim to aten"

[jit][redo] move casting ops from prim to aten

Summary:

This is a redo PR of https://github.com/pytorch/pytorch/pull/21002

Currently we are messed up with prim and aten namespace when we register prim and builtin ops, lots of the ops that have schema should go into the aten namespace rather than prim, prim namespace should only be reserved for the ops that does not have a schema.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22275 gh/wanchaol/27/head

wanchaol committed:
rebase and address comments on "[jit][redo] move casting ops from prim to aten"

[jit][redo] move casting ops from prim to aten

Summary:

This is a redo PR of https://github.com/pytorch/pytorch/pull/21002

Currently we are messed up with prim and aten namespace when we register prim and builtin ops, lots of the ops that have schema should go into the aten namespace rather than prim, prim namespace should only be reserved for the ops that does not have a schema.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22275 gh/wanchaol/27/head
